db/schema.rb

```diff
@@ -10,12 +10,33 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_04_23_153710) do
+ActiveRecord::Schema.define(version: 2020_12_23_161937) do
 
   # These are extensions that must be enabled in order to support this database
   enable_extension "plpgsql"
   enable_extension "uuid-ossp"
 
+  create_table "active_storage_attachments", force: :cascade do |t|
+    t.string "name", null: false
+    t.string "record_type", null: false
+    t.bigint "record_id", null: false
+    t.bigint "blob_id", null: false
+    t.datetime "created_at", null: false
+    t.index ["blob_id"], name: "index_active_storage_attachments_on_blob_id"
+    t.index ["record_type", "record_id", "name", "blob_id"], name: "index_active_storage_attachments_uniqueness", unique: true
+  end
+
+  create_table "active_storage_blobs", force: :cascade do |t|
+    t.string "key", null: false
+    t.string "filename", null: false
+    t.string "content_type"
+    t.text "metadata"
+    t.bigint "byte_size", null: false
+    t.string "checksum", null: false
+    t.datetime "created_at", null: false
```

Comment:
are we missing a `service_name` column as mentioned in the [comment below](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/3702/files#r575083460)

Reply:
I was unaware of the `service_name` column but it appears to be introduced in Rails 6.1:

https://guides.rubyonrails.org/6_1_release_notes.html#active-storage-notable-changes

and will be added in the normal upgrade process.

Reply:
I tried updating Rails to 6.1 to confirm this (and the other comment about the active_storage_variant_records  table). After running the `active_storage:update` task I get the following two migrations created:

`db/migrate/20210212095632_add_service_name_to_active_storage_blobs.active_storage.rb`:

```ruby
# This migration comes from active_storage (originally 20190112182829)
class AddServiceNameToActiveStorageBlobs < ActiveRecord::Migration[6.0]
  def up
    unless column_exists?(:active_storage_blobs, :service_name)
      add_column :active_storage_blobs, :service_name, :string

      if configured_service = ActiveStorage::Blob.service.name
        ActiveStorage::Blob.unscoped.update_all(service_name: configured_service)
      end

      change_column :active_storage_blobs, :service_name, :string, null: false
    end
  end

  def down
    remove_column :active_storage_blobs, :service_name
  end
end
```

`db/migrate/20210212095633_create_active_storage_variant_records.active_storage.rb`:

```ruby
# This migration comes from active_storage (originally 20191206030411)
class CreateActiveStorageVariantRecords < ActiveRecord::Migration[6.0]
  def change
    create_table :active_storage_variant_records do |t|
      t.belongs_to :blob, null: false, index: false
      t.string :variation_digest, null: false

      t.index %i[ blob_id variation_digest ], name: "index_active_storage_variant_records_uniqueness", unique: true
      t.foreign_key :active_storage_blobs, column: :blob_id
    end
  end
end
```

Reply:
I thought we had bumped to rails 6.1, but that was VCD. I guess the question becomes more or should we upgrade to rails 6.1 first then? especially as it means more rework for actives storage if we do this now

Reply:
I would be open to upgrading to 6.1 first, especially as the Variant Records (which I did not know about) looks like something we could use for the PDF document previews (although it looks to be geared primarily at resizing images).

Reply:
after having tried the 6.1 bump and it hitting various issues off the blocks. I am going to approve this for merging now. If you want to PR 6.1 before then I will leave it up to you. I am not going to have time this sprint.